### PR TITLE
Add dummy parser counter support

### PR DESCRIPTION
**Pull request by RabbitWhite1**
1. With dummy ParserCounter, the error becomes "We currently do not support parser loops". 
2. `p40f_tofino.p4` will timeout. This should be the problem of analyzing lots of `Parsing_Part1` and `Parsing_Part2`, because commenting out some of them can help.

@fruffy Any suggestions on these two problems?

### CODE REVIEW

**Comment by fruffy** [comment]
Regarding 1. yes I have not implemented a way to model loops in the parser right now. Technically we can unroll them, but it is tricky to do correctly. 

With respect to `p40f_tofino` how long does it take to analyze? 10 minutes like the big Tofino programs or a verry long time?

**Comment by RabbitWhite1** [comment]
> With respect to `p40f_tofino` how long does it take to analyze? 10 minutes like the big Tofino programs or a verry long time?

It takes very long time. I haven't tried to wait until it is completed, but it takes at least hours.

**Comment by fruffy** [comment]
Then let's comment/skip it for now. I will try to take a closer look at it next week.

**Comment by RabbitWhite1** [comment]
Sounds good. Thank you!

**Comment by RabbitWhite1** [comment]
The programs stuck after printing "Running analysis...", so I think it is the IR conversion.

**Comment by RabbitWhite1** [comment]
Okay, then I will put a patch that decrease the scale of program to pass the CI test and add annotation for this.

File: targets/tofino/base/expression_resolver.cpp [hunk 1]
```diff
@@ -143,8 +143,8 @@ const ExternMethodImpls EXTERN_METHOD_IMPLS(
      {"ParserCounter.set",
       {"value"},
       [](const ExternMethodImpls::ExternInfo &externInfo) {
-          P4C_UNIMPLEMENTED("Unimplemented extern method: %1%.%2%",
-                            externInfo.externObjectRef.toString(), externInfo.methodName);
+          //   P4C_UNIMPLEMENTED("Unimplemented extern method: %1%.%2%",
```
Comment (fruffy) [inline targets/tofino/base/expression_resolver.cpp:146]: Just remove the comment, or add a warning?

File: targets/tofino/base/expression_resolver.cpp [hunk 2]
```diff
@@ -163,36 +163,31 @@ const ExternMethodImpls EXTERN_METHOD_IMPLS(
...
      /// @return true if counter value is negative.
      {"ParserCounter.is_negative",
       {},
       [](const ExternMethodImpls::ExternInfo &externInfo) {
-          P4C_UNIMPLEMENTED("Unimplemented extern method: %1%.%2%",
-                            externInfo.externObjectRef.toString(), externInfo.methodName);
-          return nullptr;
+          auto label = externInfo.externObjectRef.path->toString() + "_" + externInfo.methodName +
+                       "_" + std::to_string(externInfo.originalCall.clone_id);
+          const auto isNegative =
```
Comment (fruffy) [inline targets/tofino/base/expression_resolver.cpp:177]: You can just return the variaable here instead of defining a new one.

**Review by fruffy — APPROVED** [review]

**Comment by RabbitWhite1** [comment]
@fruffy I re-disabled the scion.p4 due to timeout. I guess this is the same issue with p40f_tofino.p4.

**Pull request merged**